src/core/syscalls/syscall_handler.rs

```diff
@@ -13,7 +13,7 @@ pub(crate) trait SyscallHandler {
     fn call_contract(
         &mut self,
         vm: &mut VirtualMachine,
-        syscall_ptr: Relocatable,
+        request: SyscallRequest,
```

Comment:
Not sure what @mmsc2 's idea was but it seems we could pass the request struct itself instead of the enum but in any case it will be easy to normalize on my future `syscall()` PR